File: sdks/python/src/raven/resources/metrics.py
"""Real-only aggregate metrics (Phase 9) — this is also today's "usage" view
(no separate billing/usage-metering system exists yet, see
docs/sdk/server/python.md#usage).
"""

from __future__ import annotations

from typing import cast

from .._http import RavenHttpClient
from .._async_http import AsyncRavenHttpClient
from .._types import MetricsOverview, MetricsRange


class MetricsResource:
    def __init__(self, http: RavenHttpClient) -> None:
        self._http = http

    def get(self, range: MetricsRange | None = None) -> MetricsOverview:
        return cast(MetricsOverview, self._http.request("/v1/metrics", query={"range": range}))


class AsyncMetricsResource:
    def __init__(self, http: AsyncRavenHttpClient) -> None:
        self._http = http

    async def get(self, range: MetricsRange | None = None) -> MetricsOverview:
        return cast(MetricsOverview, await self._http.request("/v1/metrics", query={"range": range}))

File: packages/server-sdk/src/resources/metrics.ts
import type { RavenHttpClient } from '../http-client';
import type { MetricsOverview, MetricsRange } from '../types';

/**
 * Real aggregate metrics only (Phase 9). This doubles as today's "usage"
 * view, since no separate billing or usage-metering system exists yet. See
 * docs/sdk/server/typescript.md#usage.
 */
export class MetricsResource {
  constructor(private readonly http: RavenHttpClient) {}

  get(range?: MetricsRange): Promise<MetricsOverview> {
    return this.http.request<MetricsOverview>('/v1/metrics', { query: { range } });
  }
}
